jdaviz/configs/mosviz/plugins/viewers.py

```diff
@@ -141,6 +143,80 @@ def __init__(self, *args, **kwargs):
         # Setup viewer option defaults
         self.state.aspect = 'auto'
 
+        if self.jdaviz_app.config == 'specviz2d':
+            self.label_mouseover = None
+            self.add_event_callback(self.on_mouse_or_key_event,
+                                    events=['mousemove', 'mouseenter', 'mouseleave'])
+
+    def on_mouse_or_key_event(self, data):
```

Comment:
I added this for my own sanity here though it is technically out of scope as per original request.

Reply:
Is there anything modified from the two other instances of this (imviz and cubeviz)?  I really think we should refactor and move the generic logic into either the plugin (preferred, imo) or a shared viewer class.

Reply:
Yes, for this one, I don't think we can ever render sky coordinates. So if you want it to look pretty, it would be actually a new plugin without the sky coordinates field in vue template, or something.

Imviz has to worry about multiple 2D images. Cubeviz has to worry about 3D cube with 2D overlays. And Specviz2d has to worry about 2D spectra. So, everyone is a little different.

Reply:
> Why is info bar cut off a little?

I am guessing the coordinates display is assuming the taller app-toolbar height from imviz and cubeviz, but I don't think specviz2d warrants the increased height unless its going to use all the space.  I'd vote for this to be moved to a separate PR then so we can try to generalize it and have the displayed info be responsive to the type of underlying data.  But we can also have refactoring that as a follow-up effort if you'd prefer and can find an easy way to at least generalize the height.

Reply:
If there is strong opposition to do this now, I can take it back out but not till after we know we fixed the parsing for sure because having this is nice for sanity check.

Reply:
Cami wants this to be kept, so I opened https://github.com/spacetelescope/jdaviz/issues/1645 as a follow-up issue w.r.t. refactoring. Thanks!